### PR TITLE
Deploy game server

### PR DESCRIPTION
**Pull request by Waldz**

### CODE REVIEW

File: ci.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"github.com/mysteriumnetwork/winter-server/core"
```
Comment (soffokl) [inline ci.go:4]: It looks not formatted with a `goimports`

File: ci.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+		doRelease(os.Args[2:])
+	case "help":
+		fallthrough
+	default:
+		fmt.Print(help)
+	}
+}
+
+func main() {
+	do(os.Args[1:])
```
Comment (soffokl) [inline ci.go:188]: I'd replace `do` function by the `main` since `main` call only `do`.

File: ci.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	switch strings.ToLower(args[0]) {
+	case "test":
+		core.Execute("go test ./...")
+	case "run":
+		doRun(args[1:])
+	case "build":
+		doBuild(args[1:])
+	case "release":
+		doRelease(os.Args[2:])
+	case "help":
```
Comment (soffokl) [inline ci.go:180]: We can just remove this option, it will be done via `default` anyway.

File: ci.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+	switch strings.ToLower(args[0]) {
+	case "test":
+		core.Execute("go test ./...")
+	case "run":
+		doRun(args[1:])
+	case "build":
+		doBuild(args[1:])
+	case "release":
+		doRelease(os.Args[2:])
```
Comment (soffokl) [inline ci.go:179]: `doRelease(args[1:])`?

File: ci.go [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+		core.ExecuteIn("./server", "./out")
+	case "client":
+		buildClient()
+		core.ExecuteIn("./client", "./out")
+	case "web":
+		buildWeb()
+		core.ExecuteIn("./web", "./out")
+	case "help":
+		fmt.Print(help)
+	default:
```
Comment (soffokl) [inline ci.go:93]: Should the `doRun` have the default option that only builds everything and run nothing?

File: ci.go [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+	fmt.Print("Docker image building process complete!")
+}
+
+func publishDocker(sourceImage, targetImage string, targetVersions []string) {
+	for _, version := range targetVersions {
+		targetImageWithVersion := fmt.Sprintf("%s:%s", targetImage, version)
+		fmt.Printf("Publishing image version '%s'..\n", targetImageWithVersion)
+
+		core.MustRunCommand("docker", "tag", sourceImage, targetImageWithVersion)
+		core.MustRunCommand("docker","push", targetImageWithVersion)
```
Comment (soffokl) [inline ci.go:59]: This line looks not gofmt-ed

File: docker/server/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,12 @@
+FROM golang:1.10.2-alpine AS builder
```
Comment (soffokl) [inline docker/server/Dockerfile:1]: Maybe golang 1.10.4 or 1.11?

**Review by soffokl — COMMENTED** [review]
Looks like this repo wants to get some `gofmt`. )

**Comment by donce** [comment]
It's hard to understand what's happening here - could you add a short summary to PR description? That would be very helpful :)

**Review by mipo47 — APPROVED** [review]

**Comment by Waldz** [comment]
This game server is not where innovate.
I just wanted to communicate that I deploy it fo candidates

**Pull request merged**